### PR TITLE
Update androidTests to use non-deprecated API

### PR DESCRIPTION
**Pull request by ericcornelissen**
- Relevant Issues: #77
- Related Pull Requests: -

* * *

## What
This PR updates the androidTest test suites to move away from the deprecated `ActivityInstrumentationTestCase2` and use the `@Rule ActivityTestRule` instead.

## Why
This Pull Request is needed because we don't want to use deprecated APIs. Also, using this newer API should reduce the number of invalid failing Travis builds as it loads Activities more efficiently (but I can't promise anything 😄)

## How
This feature can be viewed/tested within the project by running the androidTests on an emulator or mobile phone and see that they still work.

## Alternative implementation
None

## Notes
The Bluetooth/connection steps have been removed due to complications with the having to enable Bluetooth in order for these tests to run properly, also they don't provide any additional coverage as the Bluetooth classes are excluded from the coverage report.

### CODE REVIEW

File: config/quality.gradle [hunk 1]
```diff
@@ -9,6 +9,27 @@ check.dependsOn 'checkstyle', 'findbugs', 'pmd', 'lint'
 def configDir = "${project.rootDir}/config/quality"
 def reportsDir = "${project.buildDir}/reports"
 
+
+checkstyle {
```
Comment (TheBonheurs) [inline config/quality.gradle]: Why change the position btw?

Comment (ericcornelissen) [inline config/quality.gradle]: Was just messing around to get rid of PMD/Findbugs errors for the androidTests ended up reordering this file. So now the plugin configurations come before the tasks...

File: app/src/androidTest/java/com/nervousfish/nervousfish/test/ContactSteps.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.intent.Intents.intended;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static junit.framework.Assert.assertFalse;
+import static junit.framework.Assert.assertTrue;
+
+@CucumberOptions(features = "features")
+public class ContactSteps {
```
Comment (clenrock) [inline app/src/androidTest/java/com/nervousfish/nervousfish/test/ContactSteps.java]: Do these name changes fix the tests, or are they just to make it neater?

Comment (ericcornelissen) [inline app/src/androidTest/java/com/nervousfish/nervousfish/test/ContactSteps.java]: That is just for consistency sake 😇

**Pull request merged**